From ea27820f0b23e99fd513370a0f1dc9eed4883adf Mon Sep 17 00:00:00 2001
From: Stefan Roese <sr@denx.de>
Date: Mon, 3 Dec 2018 10:54:57 +0100
Subject: [PATCH] net: ethernet: ralink: mtk_eth_soc: Set DMA masks

Without setting the DMA masks, the DMA subsystem complains and xfer
does not work correctly. So lets set the DMA masks in the probe
function.

Signed-off-by: Stefan Roese <sr@denx.de>
Upstream-Status: Pending [to OpenWrt]
---
 drivers/net/ethernet/ralink/mtk_eth_soc.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/net/ethernet/ralink/mtk_eth_soc.c b/drivers/net/ethernet/ralink/mtk_eth_soc.c
index 9f3da70650c6..5c258281d350 100644
--- a/drivers/net/ethernet/ralink/mtk_eth_soc.c
+++ b/drivers/net/ethernet/ralink/mtk_eth_soc.c
@@ -1627,6 +1627,19 @@ static int fe_probe(struct platform_device *pdev)
 	netif_napi_add_weight(netdev, &priv->rx_napi, fe_poll, napi_weight);
 	fe_set_ethtool_ops(netdev);
 
+	/* Init DMA mask */
+	netdev->dev.dma_mask = &netdev->dev.coherent_dma_mask;
+	err = dma_set_mask(&netdev->dev, DMA_BIT_MASK(32));
+	if (err) {
+		dev_err(&pdev->dev, "error setting DMA mask\n");
+		return err;
+	}
+	err = dma_set_coherent_mask(&netdev->dev, DMA_BIT_MASK(32));
+	if (err) {
+		dev_err(&pdev->dev, "error setting coherent DMA mask\n");
+		return err;
+	}
+
 	err = devm_register_netdev(&pdev->dev, netdev);
 	if (err) {
 		dev_err(&pdev->dev, "error bringing up device\n");
-- 
2.43.0

